components/Article/Codebox/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { highlight, languages } from 'prismjs';
+import { sanitize } from 'isomorphic-dompurify';
+import classnames from 'classnames';
+import styles from './index.module.scss';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import type { FC, PropsWithChildren, ReactElement, MouseEvent } from 'react';
+
+type CodeBoxProps = {
+  children: ReactElement<PropsWithChildren<{ className?: string }>>;
+};
+
+export const replaceLabelLanguages = (language: string) =>
+  language.replace(/console/i, 'bash');
+
+export const replaceLanguages = (language: string) =>
+  language
+    .replace(/mjs|cjs|javascript/i, 'js')
+    .replace(/console|shell/i, 'bash');
+
+const Codebox: FC<CodeBoxProps> = ({ children: { props } }) => {
+  const [parsedCode, setParsedCode] = useState('');
+  const [copied, copyText] = useCopyToClipboard();
+  const [langIndex, setLangIndex] = useState(0);
+
+  const className = props.className || 'text';
+
+  // Language Matches in class
+  const matches = className.match(/language-(?<lang>.*)/);
+  const languageOptions = (matches?.groups?.lang || 'text').split('|');
+  const language = languageOptions[langIndex];
+
+  const codeArray = props.children
+    ? props.children.toString().split('--------------\n')
+    : [''];
+
+  const handleCopyCode = async (event: MouseEvent<HTMLButtonElement>) => {
```

Comment:
This doesn't need to be `async`. There's no async operation happening here.

Reply:
Actually, `copyText` is a async operation. Maybe an `await` here would be a better option?

Reply:
Well that makes sense. But what will the onClick operator await for? Isn't onClick itself synchronous? (If I recall it does indeed in some certain blocking operations wait for something actually)